### PR TITLE
Fixes #15078: Add an option to fail or not during policy generation if NodeConfigurations cannot be created

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/15078

### CODE REVIEW

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
this does not work (or the parent ticket)

**Comment by ncharles** [comment]
Commit modified

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala [hunk 1]
```diff
@@ -119,11 +119,11 @@ case class Parallelism(
   , semaphore: TaskSemaphore
 )
 
-object ParallelSequence {
+object ParallelSequence extends Loggable {
   private[this] def boxToEither[U,T](f: U => Box[T])(u:U): Either[String, T] = f(u) match {
     case Full(t)    => Right(t)
-    case Empty      => Left("Error (no message available)")
-    case f: Failure => Left(f.messageChain)
+    case Empty      => logger.error("Empty botToEither: no result") ; Left("Error (no message available)")
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala]: bo_x_ToEither

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/appconfig/ConfigService.scala [hunk 2]
```diff
@@ -393,7 +397,7 @@ class LDAPBasedConfigService(
        rudder.save.db.compliance.levels=true
        rudder.save.db.compliance.details=false
        rudder.generation.max.parallelism=x0.5
-       rudder.generation.js.timeout=5
```
Comment (fanf) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/appconfig/ConfigService.scala]: why do you remove it?

Comment (ncharles) [reply webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/appconfig/ConfigService.scala]: it was a mistake

**Comment by ncharles** [comment]
Commit modified

**Comment by fanf** [comment]
OK, merging this PR

**Pull request merged**